test/testprocess.c

```diff
@@ -0,0 +1,666 @@
+    SDLTest_AssertCheck(result, "SDL_WriteProcess()");
+    if (!result) {
+        goto failed;
+    }
+
+    exit_code = 0xdeadbeef;
+    buffer = (char *)SDL_ReadProcess(process, NULL, &exit_code);
+    SDLTest_AssertCheck(buffer != NULL, "SDL_ReadProcess()");
+    SDLTest_AssertCheck(exit_code == 0, "Exit code should be 0, is %d", exit_code);
+    if (!buffer) {
+        goto failed;
+    }
+
+    SDLTest_Log("Text read from subprocess: %s", buffer);
+    SDLTest_AssertCheck(SDL_strcmp(buffer, text_in) == 0, "Subprocess stdout should match text written to stdin");
+    SDL_free(buffer);
+    SDLTest_AssertPass("About to destroy process");
+    SDL_DestroyProcess(process);
+    return TEST_COMPLETED;
+
+failed:
+    SDL_DestroyProcess(process);
+    return TEST_ABORTED;
+}
+
+static int process_testMultiprocessStdinToStdout(void *arg)
+{
+    TestProcessData *data = (TestProcessData *)arg;
+    const char *process_args[] = {
+        data->childprocess_path,
+        "--stdin-to-stdout",
+        NULL,
+    };
+    SDL_Process *process1 = NULL;
+    SDL_Process *process2 = NULL;
+    SDL_PropertiesID props;
+    const char *text_in = "Tests whether we can write to stdin and read from stdout\r\n{'succes': true, 'message': 'Success!'}\r\nYippie ka yee\r\nEOF";
+    char *buffer;
+    SDL_bool result;
+    int exit_code;
+
+    process1 = SDL_CreateProcess(process_args, SDL_TRUE);
+    SDLTest_AssertCheck(process1 != NULL, "SDL_CreateProcess()");
+    if (!process1) {
+        goto failed;
+    }
+
+    props = SDL_CreateProperties();
+    SDL_SetPointerProperty(props, SDL_PROP_PROCESS_CREATE_ARGS_POINTER, (void *)process_args);
+    SDL_SetNumberProperty(props, SDL_PROP_PROCESS_CREATE_STDIN_NUMBER, SDL_PROCESS_STDIO_REDIRECT);
+    SDL_SetPointerProperty(props, SDL_PROP_PROCESS_CREATE_STDIN_POINTER, SDL_GetPointerProperty(SDL_GetProcessProperties(process1), SDL_PROP_PROCESS_STDOUT_POINTER, NULL));
+    SDL_SetNumberProperty(props, SDL_PROP_PROCESS_CREATE_STDOUT_NUMBER, SDL_PROCESS_STDIO_APP);
+    process2 = SDL_CreateProcessWithProperties(props);
+    SDL_DestroyProperties(props);
+    SDLTest_AssertCheck(process2 != NULL, "SDL_CreateProcess()");
+    if (!process2) {
+        goto failed;
+    }
+
+    SDLTest_AssertPass("About to write to process");
+    result = SDL_WriteProcess(process1, text_in, SDL_strlen(text_in), SDL_TRUE);
+    SDLTest_AssertCheck(result, "SDL_WriteProcess()");
+    if (!result) {
+        goto failed;
+    }
+
+    exit_code = 0xdeadbeef;
+    buffer = (char *)SDL_ReadProcess(process2, NULL, &exit_code);
+    SDLTest_AssertCheck(buffer != NULL, "SDL_ReadProcess()");
+    SDLTest_AssertCheck(exit_code == 0, "Exit code should be 0, is %d", exit_code);
+    if (!buffer) {
+        goto failed;
+    }
+
+    SDLTest_Log("Text read from subprocess: %s", buffer);
+    SDLTest_AssertCheck(SDL_strcmp(buffer, text_in) == 0, "Subprocess stdout should match text written to stdin");
```

Comment:
Rebasing on top of current master will give you `SDLTest_LogEscapedString`.
Allowing to diagnose issues with non-printable charaters

```suggestion
    buffer = (char *)SDL_ReadProcess(process2, &bytes_read, &exit_code);
    SDLTest_AssertCheck(buffer != NULL, "SDL_ReadProcess()");
    SDLTest_AssertCheck(exit_code == 0, "Exit code should be 0, is %d", exit_code);
    if (!buffer) {
        goto failed;
    }

    SDLTest_LogEscapedString("Expected text read from subprocess: ", text_in, SDL_strlen(bytes_read));
    SDLTest_LogEscapedString("Actual text read from subprocess: ", buffer, bytes_read);
    SDLTest_AssertCheck(bytes_read == SDL_strlen(text_in), "Expected to read %u bytes, actual %u bytes", (unsigned) SDL_strlen(text_in), bytes_read);
    SDLTest_AssertCheck(SDL_strcmp(buffer, text_in) == 0, "Subprocess stdout should match text written to stdin");
```

Reply:
This doesn't seem right:
`SDL_strlen(bytes_read)`
We can make this change after this PR merges. I don't want to complicate the process.

Reply:
>  I don't want to complicate the process.

I see what you did there :)